File: fopa_sop_apk/lib/features/feat_user/domaines/dtos/update_user_dto.dart
import 'package:fopa_sop_apk/features/feat_auth/datas/models/user_model.dart';

class UpdateUserDto {
  final String? email;
  final String? name;
  final String? phone;
  final String? username;
  final String? password;
  final UserRole? role;
  final bool? isActive;

  UpdateUserDto({
    this.email,
    this.name,
    this.phone,
    this.username,
    this.password,
    this.role,
    this.isActive,
  });

  Map<String, dynamic> toJson() {
    final map = <String, dynamic>{};
    if (email != null) map['email'] = email;
    if (name != null) map['name'] = name;
    if (phone != null) map['phone'] = phone;
    if (username != null) map['username'] = username;
    if (password != null) map['password'] = password;
    if (role != null) map['role'] = role!.name;
    if (isActive != null) map['isActive'] = isActive;
    return map;
  }
}

File: src/application/dtos/user/update-user.dto.ts
import { ApiProperty } from '@nestjs/swagger';
import { IsEmail, IsEnum, IsOptional, IsString, MinLength } from 'class-validator';
import { UserRole } from '../../../domain/entities/user.entity';

export class UpdateUserDto {
  @ApiProperty({ required: false })
  @IsOptional()
  @IsEmail({}, { message: "L'email doit être valide" })
  email?: string;

  @ApiProperty({ required: false })
  @IsOptional()
  @IsString()
  name?: string;

  @ApiProperty({ required: false })
  @IsOptional()
  @IsString()
  phone?: string;

  @ApiProperty({ required: false })
  @IsOptional()
  @IsString()
  username?: string;

  @ApiProperty({ required: false, minLength: 6 })
  @IsOptional()
  @IsString()
  @MinLength(6, { message: 'Le mot de passe doit contenir au moins 6 caractères' })
  password?: string;

  @ApiProperty({ enum: UserRole, required: false })
  @IsOptional()
  @IsEnum(UserRole, { message: 'Le rôle doit être ADMIN, CAISSIERE ou CLIENT' })
  role?: UserRole;

  @ApiProperty({ required: false })
  @IsOptional()
  isActive?: boolean;
}
